src/ocsp.c

```diff
@@ -1074,34 +1074,39 @@ WOLFSSL_OCSP_CERTID* wolfSSL_d2i_OCSP_CERTID(WOLFSSL_OCSP_CERTID** cidOut,
 {
     WOLFSSL_OCSP_CERTID *cid = NULL;
 
-    if ((cidOut == NULL) || (derIn == NULL) || (length == 0))
-        goto err;
+    if ((cidOut != NULL) && (derIn != NULL) && (length > 0)) {
+
+        cid = *cidOut;
+
+        /* If a NULL is passed we allocate the memory for the caller. */
+        if (cid == NULL)
```

Comment:
Please always add {} for ifs. I know some people don't, but we really should. It's very easy to have a bad bug where someone adds another line below the one incumbent line in a conditional without {} and that new line is now outside the conditional.